src/shared/search/GlobalSearch.tsx

```diff
@@ -0,0 +1,82 @@
+import React, {FC, useEffect, memo, useState, useMemo, useCallback} from 'react'
+import {useLocation} from 'react-router-dom'
+import {DocSearchModal} from '@docsearch/react'
+import {v4 as uuid} from 'uuid'
+import {ClickOutside} from 'src/shared/components/ClickOutside'
+import '@docsearch/react/style'
+
+import './GlobalSearch.scss'
+
+import {event} from 'src/cloud/utils/reporting'
+
+const Hit: FC<any> = memo(({hit, children}) => {
+  const linkID = uuid()
+  const location = useLocation()
+  const logClickedSearchQuery = useCallback(() => {
+    const clickedElem = document.getElementById(linkID) as HTMLLinkElement
+    const searchQuery = document.getElementById(
+      'docsearch-input'
+    ) as HTMLInputElement
+    event(
+      'clicked search result',
+      {
+        resultURL: clickedElem.href,
+        fromPage: location.pathname,
+        query: searchQuery?.value,
+      },
+      {}
+    )
+  }, [linkID, location.pathname])
```

Comment:
should we be concerned about having a `uuid()` in the deps array? Feel like that's something that might change considering line 13?

Reply:
the reason its in here is to track that change

Reply:
sorry maybe I'm not quite following - but wouldn't `linkID` change every time this is rendered? So it will basically always re-render and will never stay memoized?